internal/binder/function/funcs_array.go

```diff
@@ -105,4 +111,327 @@ func registerArrayFunc() {
+		},
+		val: func(ctx api.FunctionContext, args []ast.Expr) error {
+			return ValidateLen(2, len(args))
+		},
+	}
+
+	builtins["array_last_position"] = builtinFunc{
+		fType: ast.FuncTypeScalar,
+		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
+			array, ok := args[0].([]interface{})
+			if !ok {
+				return errorArrayFirstArgumentNotArrayError, false
+			}
+
+			for i := len(array) - 1; i >= 0; i-- {
+				if array[i] == args[1] {
+					return i + 1, true
+				}
+			}
+
+			return 0, true
+		},
+		val: func(ctx api.FunctionContext, args []ast.Expr) error {
+			return ValidateLen(2, len(args))
+		},
+	}
+
+	builtins["array_contains_any"] = builtinFunc{
+		fType: ast.FuncTypeScalar,
+		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
+			array1, ok1 := args[0].([]interface{})
+			if !ok1 {
+				return errorArrayFirstArgumentNotArrayError, false
+			}
+			array2, ok2 := args[1].([]interface{})
+			if !ok2 {
+				return errorArraySecondArgumentNotArrayError, false
+			}
+
+			for _, a := range array1 {
+				for _, b := range array2 {
+					if a == b {
+						return true, true
+					}
+				}
+			}
+
+			return false, true
+		},
+		val: func(ctx api.FunctionContext, args []ast.Expr) error {
+			return ValidateLen(2, len(args))
+		},
+	}
+
+	builtins["array_intersect"] = builtinFunc{
+		fType: ast.FuncTypeScalar,
+		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
+			array1, ok1 := args[0].([]interface{})
+			if !ok1 {
+				return errorArrayFirstArgumentNotArrayError, false
+			}
+			array2, ok2 := args[1].([]interface{})
+			if !ok2 {
+				return errorArraySecondArgumentNotArrayError, false
+			}
+			intersection := []interface{}{}
+			set := make(map[interface{}]bool)
+
+			for _, a := range array1 {
+				set[a] = true
+			}
+
+			for _, b := range array2 {
+				if set[b] {
+					intersection = append(intersection, b)
+					set[b] = false
+				}
+			}
+
+			return intersection, true
+		},
+		val: func(ctx api.FunctionContext, args []ast.Expr) error {
+			return ValidateLen(2, len(args))
+		},
+	}
+
+	builtins["array_union"] = builtinFunc{
+		fType: ast.FuncTypeScalar,
+		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
+			array1, ok1 := args[0].([]interface{})
+			if !ok1 {
+				return errorArrayFirstArgumentNotArrayError, false
+			}
+			array2, ok2 := args[1].([]interface{})
+			if !ok2 {
+				return errorArraySecondArgumentNotArrayError, false
+			}
+			union := []interface{}{}
+			set := make(map[interface{}]bool)
+
+			for _, a := range array1 {
+				if !set[a] {
+					union = append(union, a)
+					set[a] = true
+				}
+			}
+			for _, b := range array2 {
+				if !set[b] {
+					set[b] = true
+					union = append(union, b)
+				}
+			}
+
+			return union, true
+		},
+		val: func(ctx api.FunctionContext, args []ast.Expr) error {
+			return ValidateLen(2, len(args))
+		},
+	}
+	builtins["array_max"] = builtinFunc{
```

Comment:
It seems `array_max` has lots of duplicated codes with `array_min`, could u remove the duplicated code and make it easy and clear?

Reply:
I ran into some function boundary handling details during the implementation, considering that the ARRAY_CREATE function allows elements of arrays to have different data types. Specifically, the following: Does the function also need to cover numbers in the form of strings, such as "123"? Is it possible to have an array of mixed strings, integers, and floats?

Reply:
> Is it possible to have an array of mixed strings, integers, and floats?

I think it's possible.